cmd/snap-exec/main.go

```diff
@@ -253,9 +252,16 @@ func execHook(snapName, revision, hookName string) error {
 	}
 
 	// build the environment
-	env := append(os.Environ(), osutil.SubstituteEnv(hook.Env())...)
+	env, err := osutil.OSEnvironment()
+	if err != nil {
+		return err
+	}
+	// NOTE: we do not do env.UnescapeSaved()
```

Comment:
this is probably not intended, otoh given where hooks are run from for now, it's not material, symmmetry is probably preferable though